server/lib.go

```diff
@@ -13,9 +13,10 @@ import (
 	"github.com/mochi-co/mqtt/v2/listeners"
 )
 
-var tcpPort = flag.Int("tcp-port", 1883, "TCP port number to listen on")
+var PORT = 1883
 
-func main() {
+func Serve() {
+	var tcpPort = flag.Int("tcp-port", PORT, "TCP port number to listen on")
```

Comment:
For some reason, I see in most projects the flag is declared globally. That's why I kept it there. Pls verify again else leave it's alright

Reply:
I doubt you can compile if there is no main function. The server code should be independent and iiuc should have its own main as starting point

Reply:
I thought let is have `Serve` here and in the `cmd/chaTer-server/main.go` we can call this `Serve()` functions with arguments passed in through `CLI`. Will be easy to deploy also.

What do you think ?

Reply:
Why are we typing up the CLI with server? Do we need a CLI interface at all for the server. I was thinking we build the server separately and deploy it somewhere (container mostly) on some port. That's all. What do you want to do with a server CLI?

Even if we need a server CLI, I think the directory structure would look neater like this
- client
--- CLI
--- main.go and other files/dir
- server
---CLI
--- main.go and other files/dir

Reply:
- myproject
    - client
        - main.go
        - client.go
        - client_test.go
    - server
        - main.go
        - server.go
        - server_test.go
    - common
        - util.go
    - go.mod
    - go.sum
    - README.md

^ This will be perfect. Both main.gos are independent and it's more like the calling place. Major chunk of logic would be in server.go/client.go and we could use other relevant directories

Reply:
Made the change.